tests/services/files/test_file_service.py

```diff
@@ -450,3 +452,108 @@ def test_read_not_committed_external_file(
+    ].record_cls.deletion_status.is_deleted = True
+    recid = example_file_record["id"]
+    file_to_initialise = [
+        {
+            "key": "article.txt",
+            "checksum": "md5:c785060c866796cc2a1708c997154c8e",
+            "size": 17,  # 2kB
+            "metadata": {
+                "description": "Published article PDF.",
+            },
+        }
+    ]
+    # Initialize file saving
+    result = file_service.init_files(identity_simple, recid, file_to_initialise)
+    assert result.to_dict()["entries"][0]["key"] == file_to_initialise[0]["key"]
+    # for to_file in to_files:
+    content = BytesIO(b"test file content")
+    result = file_service.set_file_content(
+        identity_simple,
+        recid,
+        file_to_initialise[0]["key"],
+        content,
+        content.getbuffer().nbytes,
+    )
+    # TODO figure response for succesfully saved file
+    assert result.to_dict()["key"] == file_to_initialise[0]["key"]
+
+    result = file_service.commit_file(identity_simple, recid, "article.txt")
+    # TODO currently there is no status in the json between the initialisation
+    # and the commiting.
+    assert result.to_dict()["key"] == file_to_initialise[0]["key"]
+
+    # List files
+    with pytest.raises(RecordDeletedException):
+        file_service.list_files(identity_simple, recid)
+
+    with pytest.raises(RecordDeletedException):
+        file_service.list_files(identity_simple, recid, include_deleted=True)
+
+    result = file_service.list_files(superuser_identity, recid, include_deleted=True)
+    assert result.to_dict()["entries"][0]["key"] == file_to_initialise[0]["key"]
+    assert result.to_dict()["entries"][0]["storage_class"] == "L"
+    assert "uri" not in result.to_dict()["entries"][0]
+
+    # Read file metadata
+    with pytest.raises(RecordDeletedException):
+        file_service.read_file_metadata(identity_simple, recid, "article.txt")
+
+    with pytest.raises(RecordDeletedException):
+        file_service.read_file_metadata(
+            identity_simple, recid, "article.txt", include_deleted=True
+        )
+
+    result = file_service.read_file_metadata(
+        superuser_identity, recid, "article.txt", include_deleted=True
+    )
+    assert result.to_dict()["key"] == file_to_initialise[0]["key"]
+    assert result.to_dict()["storage_class"] == "L"
+    assert "uri" not in result.to_dict()
+
+    # Retrieve file
+    with pytest.raises(RecordDeletedException):
+        file_service.get_file_content(identity_simple, recid, "article.txt")
+
+    with pytest.raises(RecordDeletedException):
+        file_service.get_file_content(
+            identity_simple, recid, "article.txt", include_deleted=True
+        )
+
+    result = file_service.get_file_content(
+        superuser_identity, recid, "article.txt", include_deleted=True
+    )
+    assert result.file_id == "article.txt"
+
+    # Delete file
+    result = file_service.delete_file(identity_simple, recid, "article.txt")
```

Comment:
do we allow to an user to delete a file from deleted record? I don't think it should happen. Once the record is deleted, it should be "frozen", check if RFC has any info about this case: https://github.com/inveniosoftware/invenio-app-rdm/issues/2289

Reply:
We are using mocked [permissions](https://github.com/inveniosoftware/invenio-records-resources/blob/master/tests/mock_module/permissions.py) here, that's why a simple identity can delete files, the real permissions set for rdm should be tested in invenio-rdm-records.